gadgets/mri_core/generic_recon_gadgets/GenericReconBase.h

```diff
@@ -68,11 +72,140 @@ namespace Gadgetron {
         // exporter
         Gadgetron::ImageIOAnalyze gt_exporter_;
 
+        // store buffer names
+        std::map<std::string, std::pair<std::string, std::shared_ptr<std::ofstream> > > buffer_names_;
```

Comment:
Again, it is pretty gnarly to this as a member variable in this class and to have the low level code to manage the streams exposes throughout the Gadget.

Reply:
Move to the class in the mri_core_stream.h/cpp.